File: csharp/Validators/FileExistenceValidator.cs
using System;
using System.Collections.Generic;
using System.IO;

namespace Validators
{
    public class FileExistenceValidator: AbstractValidator
    {
        private string FilePath { get; set; }

        public FileExistenceValidator(List<string> args, ushort index, string message) : base(args, message)
        {
            FilePath = (Arguments.Count > index) ? Arguments[index] : null;
        }

        public override bool Validate()
        {
            if (FilePath == null || !File.Exists(FilePath))
            {
                Console.WriteLine(Message);
                return false;
            }

            return true;
        }
    }
}

File: python/Validators/FileExistenceValidator.py
from Abstractions.AbstractFilesystemEntryValidator import *


class FileExistenceValidator(AbstractFilesystemEntryValidator):
    def __init__(self, validate_this_file, message):
        super().__init__(validate_this_file, message)

    def validate(self):
        if not self._validate_this.exists() or not self._validate_this.is_file():
            self._print_validation_error()
            return False
        return True
